aiohttp/client.py

```diff
@@ -231,6 +222,24 @@ def _request(self, method, url, *,
         try:
             with timer:
                 while True:
+                    url, auth_from_url = strip_auth_from_url(url)
+                    if auth and auth_from_url:
+                        raise ValueError("Cannot combine AUTH argument with "
+                                         "credentials encoded in URL")
+
+                    if auth is None:
+                        auth = auth_from_url
+                    if auth is None:
+                        auth = self._default_auth
+                    # It would be confusing if we support explicit
+                    # Authorization header with auth argument
+                    if (headers is not None and
+                            auth is not None and
+                            hdrs.AUTHORIZATION in headers):
+                        raise ValueError("Cannot combine AUTHORIZATION header "
```

Comment:
Can we have tests which triggers these ValueError's? I'm hardly imagine the case for them.

Reply:
Yes, the PR has a test for the case

Reply:
Hm...seems  it's slipped away from me. Ok.